Shoko.Server/API/v3/Controllers/TreeController.cs

```diff
@@ -88,27 +83,34 @@ public ActionResult<ListResult<Group>> GetFilteredGroups([FromRoute] int filterI
 
                 groups = groupIds
                     .Select(group => RepoFactory.AnimeGroup.GetByID(group))
-                    .Where(group => group != null)
+                    .Where(group =>
+                    {
+                        if (group == null || group.AnimeGroupParentID.HasValue)
+                            return false;
+
+                        return includeMissing || group.GetAllSeries().Any(s => s.GetAnimeEpisodes().Any(e => e.GetVideoLocals().Count > 0));
+                    })
                     .OrderByGroupFilter(groupFilter);
             }
 
             return groups
-                .ToListResult(group => new Group(HttpContext, group), page, pageSize);
+                .ToListResult(group => new Group(HttpContext, group, randomImages), page, pageSize);
         }
 
         /// <summary>
         /// Get a list of all the sub-<see cref="Group"/>s belonging to the <see cref="Group"/> with the given <paramref name="groupID"/> and which are present within the <see cref="Filter"/> with the given <paramref name="filterID"/>.
         /// </summary>
         /// <param name="filterID"><see cref="Filter"/> ID</param>
         /// <param name="groupID"><see cref="Group"/> ID</param>
+        /// <param name="includeMissing">Include <see cref="Series"/> with missing <see cref="Episode"/>s in the search.</param>
         /// <param name="randomImages">Randomise images shown for the <see cref="Group"/>.</param>
         /// <returns></returns>
         [HttpGet("Filter/{filterID}/Group/{groupID}/Group")]
-        public ActionResult<List<Group>> GetFilteredSubGroups([FromRoute] int filterID, [FromRoute] int groupID, [FromQuery] bool randomImages = false)
+        public ActionResult<List<Group>> GetFilteredSubGroups([FromRoute] int filterID, [FromRoute] int groupID, [FromQuery] bool includeMissing = false, [FromQuery] bool randomImages = false)
```

Comment:
If we are adding new, optional fields, as a general rule of thumb, add them at the end of the parameter list.

This is namely because else it can break other code when you insert a new field in the list of stuff, as using named arguments isn't always the default action - yes this is a HTTP action, but better to avoid in general.